src/plugins/keras_support/resize_operation.cpp

```diff
@@ -0,0 +1,221 @@
+            &create_resize_operation, &create_primitive<resize_operation>,
+            R"(a
+            Args:
+
+                a (array_like) : input array
+                height_factor (integer) : enlargement factor for height
+                width_factor (integer) : enlargement factor for width
+                interpolation (string) : interpolation type, "bilinear" or "nearest"
+
+            Returns:
+
+            Returns a resized version of the array with height_factor and width_factor
+            using the specified interpolation .)")};
+
+    ///////////////////////////////////////////////////////////////////////////
+    resize_operation::resize_operation(primitive_arguments_type&& operands,
+        std::string const& name, std::string const& codename)
+      : primitive_component_base(std::move(operands), name, codename)
+    {
+    }
+
+    ///////////////////////////////////////////////////////////////////////////
+    template <typename T>
+    primitive_argument_type resize_operation::nearest(ir::node_data<T>&& arg,
+        std::int64_t height_factor, std::int64_t width_factor,
+        std::string interpolation) const
+    {
+        auto m = arg.matrix();
+        blaze::DynamicMatrix<T> result(
+            m.rows() * height_factor, m.columns() * width_factor);
+
+        for (std::size_t i = 0;
+             i < m.rows() * height_factor * m.columns() * width_factor;
+             ++i)
+        {
+            std::size_t fx = i / result.columns();
+            std::size_t fy = i - fx * result.columns();
+            result(fx, fy) = m(fx / height_factor, fy / width_factor);
+        }
+
+        return primitive_argument_type{std::move(result)};
+    }
+
+    template <typename T>
+    primitive_argument_type resize_operation::bilinear(ir::node_data<T>&& arg,
+        std::int64_t height_factor, std::int64_t width_factor,
+        std::string interpolation) const
+    {
+        auto m = arg.matrix();
+        blaze::DynamicMatrix<T> result(
+            m.rows() * height_factor, m.columns() * width_factor, 0.);
+
+        for (std::size_t i = 0;
+             i < m.rows() * height_factor * m.columns() * width_factor;
+             ++i)
+        {
+            auto fx = i / result.columns();
+            auto fy = i - fx * result.columns();
+            auto ul = m(fx / height_factor, fy / width_factor);
+            auto ur = m(fx / height_factor, fy / width_factor + 1);
+            auto ll = m(fx / height_factor + 1, fy / width_factor);
+            auto lr = m(fx / height_factor + 1, fy / width_factor + 1);
+
+            if (fx / height_factor == m.rows() - 1)
+            {
+                ll = ul;
+                lr = ur;
+            }
+            if (fy / width_factor == m.columns() - 1)
+            {
+                ur = ul;
+                lr = ll;
+            }
+
+            result(fx, fy) =
+                (ul + (fy % width_factor) * (ur - ul) / width_factor) +
```

Comment:
I think `fy % width_factor` and `fx % height_factor` could be calculated once before the expression